rfd/0170-package-distribution-v3.md

```diff
@@ -0,0 +1,450 @@
+  rate of $175/hour, and an average total of 10 hours per week wasted by devs
+  due to the current solution (primarily maintenance + waiting for job
+  completion), this solution is costing $91k/year in dev time.
+  
+  AWS shows that we're storing roughly 1.25 TB of artifacts in APT and YUM S3
+  buckets. It also shows that customers download about 180 TB of artifacts every
+  month, via Cloudfront. This is increasing by approximately $800 month. The
+  total cost to us for downloads last month was $11.9k, and the total cost for
+  last year was approximately $75k. Assuming the last year's cost increase trend
+  continues, this coming year will cost us about $200k.
+
+  The total cost of this solution over the next year will _conservatively_ be
+  almost $300k. This does not take into account the (hard to quantify)
+  opportunity cost of dev time spent operating and managing the current
+  solution, which could be spent on higher-value work.
+
+One thing RFD 58 did get right is the structure of the OS package repos. This
+does not need to change, and the RFD will not discuss any changes to this.
+
+## Details
+
+Any new solution must meet the following:
+* Customers should not need to know about the new solution, or be negatively
+  impacted by it. This means no loss of availability, no regression in
+  functionality, and no required change for how customers install our products.
+  The implementation of a new solution must be _entirely_ transparent to
+  customers.
+* We must not be required to provide sensitive secrets to any new third-party
+  vendor, including signing keys for the repos metadata. These keys must never
+  leave the trusted systems that we control.
+* The new solution must meet all of our typical security requirements for
+  release pipelines (audit logging, RBAC or Teleport support, etc.) and must
+  pass an external security audit.
+* The solution must be cost effective, with a TCO less than estimates of our
+  current solution.
+* The total time to publish all the OS packages per release must be under ten
+  minutes, in an "apples to apples" comparison (same number of packages in
+  repos, same number of new packages, same number of repos, etc.)
+* The new solution must be technically simple, compared to our current solution.
+  Our current solution is tightly coupled with multiple repos, auto updates
+  tooling, GHA itself, our GHA EKS cluster, and several internal tools. It
+  should be "easy" for us to:
+  * Add and remove repos
+  * Add and remove packages/products
+  * Configure repo-type-specific properties (such as APT
+    Phased-Update-Percentage)
+* The customer-facing portion of the solution (the repos themselves) must be
+  highly available and fault-tolerant. No regular maintenance tasks (such as
+  upgrades of the solution) may result in downtime or outages of the
+  customer-facing portion. Customer-facing requests (such as those generated
```

Comment:
Nit: I would split "customer-facing requests must not run through a service" into its own bullet point - I think it's important to highlight we don't want to run an entrypoint service for downloads ourselves. Right now it's a bit buried here. Maybe even move it higher.

Reply:
Moved everything about customer impact to the top of the list, and split this as requested